marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/getOpticPlan.sjs

```diff
@@ -99,11 +100,24 @@ const qryTxt = cts.query(fn.head(searchTxtResponse));
 
 const ctsQry = cts.andQuery([qry, qryTxt]);
 
+let orderDefinitions = [];
+if (sortOrder) {
+  // convert ArrayNode to Array with .toObject()
+  for (let sort of sortOrder.toObject()) {
+    const col = op.col(sort.name);
+    if (sort.ascending) {
+      orderDefinitions.push(op.asc(col));
+    } else {
+      orderDefinitions.push(op.desc(col));
+    }
+  }
+}
 // Workaround since there's a bug in the bridge between v8 and ML builtin functions
 const qryString = xdmp.describe(ctsQry, null, null);
 const opticPlan = eval(`op.fromView(schemaName, viewName)
   .where(${qryString})
   .limit(limit)
+  ${orderDefinitions.length ? '.orderBy(orderDefinitions)': ''}
```

Comment:
@ryanjdew  @akshaysonvane  @rahulvudutala  An order definition could reference a non-existent property - e.g. Pari saves a query that references a "size" property, then Stewart removes "size" from the entity definition. How does Optic react to column names that it doesn't recognize? 

I'm wondering if a couple ml unit tests would be useful here as well to verify that the optic plan is constructed correctly based on inputs. The JUnit test is nice from an integration perspective, but there are a lot of moving parts here. A unit test could also document what happens if a non-existent column is passed in.